### PR TITLE
[Do not merge] Fix  #919061 and refactor dialog

### PR DESCRIPTION
**Pull request by ancorgs**
@kmroz contributed a fix for bug#919061 (before it was reported, actually). But that fix made the longest method become even longer, which pissed Rubocop off. I though the error was only in master, so I said "instead of increasing Rubocop limits, let's do a very small refactoring".... famous last words.

Most likely, I'll first merge only the @kmroz's fix in SLE12 and master (raising the Rubocop limits), release both and only afterwards I'll try to merge this again in master and lower the limits again (hence the "don not merge" warning). Anyways, I'm opening the pull request to gather feedback about the approach and to prove that I'm not raising the rubocop limits as a first lazy approach.

A full refactoring is not intended AT ALL, but I guess @jreidinger would like to see how the Dialog base class can be used to add some sanity to the old code.

### CODE REVIEW

**Comment by kobliha** [comment]
@ancorgs Refactoring without test case? And you want to release it as an online update?

**Comment by ancorgs** [comment]
@kobliha No, I don't want to release it as an online update. That's exactly what I was trying to explain in the (lengthly) description. First, release the simple fix (relaxing Rubocop to make it pass) as an online update. Later on, commit this refactoring only into master.

On the other hand, it's completely impossible to test the original code. I could add tests to the new code, but not use the "first write tests, then refactor" approach.

File: src/include/nfs/routines.rb [hunk 1]
```diff
@@ -37,7 +37,7 @@ def SpecToServPath(spec)
         serv = Builtins.substring(spec, 0, path_begin)
         spec = Builtins.substring(spec, Ops.add(path_begin, 1))
       end
-      term(:couple, serv, spec)
+      Term.new(:couple, serv, spec)
```
Comment (jreidinger) [inline src/include/nfs/routines.rb]: why this change is needed?

File: src/lib/nfs_client/fstab_entry_dialog.rb [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+      # help text 1/4
+      # change: locally defined -> servers on LAN
+      helptext = _(
+        "<p>Enter the <b>NFS Server Hostname</b>.  With\n" \
+          "<b>Choose</b>, browse through a list of\n" \
+          "NFS servers on the local network.</p>\n"
+      )
+      # help text 2/4
+      # added "Select" button
+      helptext << _(
```
Comment (jreidinger) [inline src/lib/nfs_client/fstab_entry_dialog.rb]: boom, crash :)
you cannot append to translated text as it is frozen so result is exception.

**Comment by jreidinger** [comment]
I agree that tests for old code can be tricky, but new dialog deserve at least trivial test that it do not crash, when you click on OK :)

File: src/include/nfs/routines.rb (hunk 1 above)
Comment (ancorgs) [inline src/include/nfs/routines.rb]: Because NfsRoutinesInclude is included in NfsClient::FstabEntryDialog, which has no access to the shortcut. Most likely, a final clean solution would not imply such thing. But this first step was intended to just reshuffle the code into a separate class, without changing the other existing pieces.

File: src/lib/nfs_client/fstab_entry_dialog.rb (hunk 2 above)
Comment (ancorgs) [inline src/lib/nfs_client/fstab_entry_dialog.rb]: Good to know. :-) Even if it actually does not crash in my system (I did quite some monkey-testing).

Comment (jreidinger) [inline src/lib/nfs_client/fstab_entry_dialog.rb]: well, it crash only if you use non-english as for english it return parameter, but if it return translated string, it is frozen

File: src/lib/nfs_client/fstab_entry_dialog.rb [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+      # popup heading
+      Yast::Popup.LongText(_("Help"), RichText(helptext), 50, 18)
+    end
+
+    protected
+
+    def create_dialog
+      super
+      Yast::UI.ChangeWidget(Id(:serverent), :Value, @server)
+      Yast::UI.SetFocus(Id(:serverent))
```
Comment (ancorgs) [inline src/lib/nfs_client/fstab_entry_dialog.rb]: @jreidinger I don't know how common is this (performing actions right after opening the dialog). It's maybe something to support in the base class somehow?

Comment (jreidinger) [inline src/lib/nfs_client/fstab_entry_dialog.rb]: how you expect to support such actions? via helpers like set_focus? I do not see big benefit and do not see better way how to support it.

Comment (ancorgs) [inline src/lib/nfs_client/fstab_entry_dialog.rb]: I don't know. Just thinking aloud. Somehow it looked to me not so different from ```dialog_options```. Maybe having a (by default empty) method ```dialog_init``` or something like that to be called at the end ```create_dialog```. Just in case is a common pattern to perform such initialization actions on the widgets. I'm fine with overloading ```create_dialog``` if it's something we are going to do rarely.

**Comment by ancorgs** [comment]
Closing. Since this was only opened to gather feedback (already provided). A real pull request will follow.
